client_plugin/drivers/vfile/vfile_driver.go

```diff
@@ -613,6 +719,15 @@ func (d *VolumeDriver) mountVFileVolume(volName string, mountpoint string, volRe
 		return err
 	}
 
+	err = d.addVMToClientList(volName, addr)
+	if err != nil {
```

Comment:
Leys say the volume was successfully mounted in the VM but updating the metadata in etcd failed. Basically the metadata does not match the reality. How are you doing to handle this scenario?

Reply:
In that case,  function "mountVFileVolume" will return with error, then in function "MountVolume", it will handle this error and mark the "volume mount failed".  Do I miss anything?

Reply:
But the volume is *actually* mounted in the case I mentioned. Dont you have to unmount it in case of failure? I dont see code that does unmount in case of failure. 

I looked at MountVolume function, I see that we decrement global refcount in case of error. What if this decrement operation fails due to network problem or etcd down etc?

Reply:
Fixed.